source/apiVolontaria/volunteer/models.py

```diff
@@ -351,3 +374,41 @@ def duration(self):
             return timedelta(minutes=self.presence_duration_minutes)
         else:
             return self.event.duration
+
+    def generate_email_invitation(self):
+        calendar = self.event.generate_ics()
+
+        if settings.CONSTANT['EMAIL_SERVICE'] is True:
+
+            email_to = self.user.email
+
+            # Debug Mode checking if we have an email in DEBUG_EMAIL_TO
+            # or disable the sending of email
+            if settings.DEBUG:
+                try:
+                    email_to = settings.DEBUG_EMAIL_TO
+                except AttributeError:
+                    email_to = None
+
+            msg = EmailMultiAlternatives(
+                subject=None,
+                body=None,
+                to=[email_to, ],
+            )
+
+            msg.attach('nousrire.ics', calendar.decode(), 'text/calendar')
+
+            msg.from_email = None
+            msg.template_id = settings.SENDINBLUE_TEMPLATE['CALENDAR_INVITATION']
+            msg.merge_global_data = {
+                'activity': self.event.task_type.name,
+                'location': self.event.cell.address.__str__(),
+                'date_month': self.event.start_date.astimezone().strftime("%m"),
+                'date_day': self.event.start_date.astimezone().strftime("%d"),
+                'time_start': self.event.start_date.astimezone().strftime("%H:%M"),
+                'time_end': self.event.end_date.astimezone().strftime("%H:%M"),
```

Comment:
As valerie said in her email it can be good to have a more comprehensible date for the user, the email system doesn't contain logic so IMHO you can just send:

 - one string who represent the full date
 - one string to represent the starttime of the event
 - one string to represent the endtime of the event

Reply:
Honestly, I really wanted to do this but I was struggling on how to send the date in french the good way ;)  Do you have any cues ?

Reply:
Habitually you will need to play with python's local : 
https://docs.python.org/3/library/locale.html#locale.setlocale

But with Django i think it already work just with this line of settings no ?:
https://github.com/Volontaria/API-Volontaria/blob/6ba0ba9d9e2f236fdafab4fd5d0300a493d4368a/source/apiVolontaria/apiVolontaria/settings.py#L123

Reply:
I have set this in my local_settings already : 
`LANGUAGE_CODE = 'fr-ca'`
And I'm getting : 
`'22 February'`
From :
`participation.event.start_date.astimezone().strftime("%d %B")`